main.c

```diff
@@ -88,14 +89,37 @@ ioline_t ledRows[NUM_ROW * 4] = {
  * Active profiles
  * Add profiles from source/profiles.h in the profile array
  */
-typedef void (*profile)( led_t*, uint8_t );
+typedef void (*lighting_callback)( led_t*, uint8_t );
+
+typedef struct {
```

Comment:
I decided to create a separate struct for holding all data relevant to the profile.
Currently it includes the callback and the timing frequency controls.
I makes the code cleaner since we would not need to check for each function as we did before.

I am planning on adding another flag `reactive` (or something similar) which would represent profiles which light up keys based on what user presses.